non-constant-expression cannot be narrowed from type 'long' to 'unsigned long' in initializer list [-Wc++11-narrowing]

seeing this building on main
```
#0 639.8   transformer_engine/jax/csrc/extensions/activation.cpp:40:45: error: non-constant-expression cannot be narrowed from type 'long' to 'unsigned long' in initializer list [-Wc++11-narrowing]
#0 639.8      40 |   auto input_shape = std::vector<size_t>{m, act_len * n};
#0 639.8         |                                             ^~~~~~~~~~~
#0 639.8   transformer_engine/jax/csrc/extensions/activation.cpp:40:45: note: insert an explicit cast to silence this issue
#0 639.8      40 |   auto input_shape = std::vector<size_t>{m, act_len * n};
#0 639.8         |                                             ^~~~~~~~~~~
#0 639.8         |                                             static_cast<unsigned long>( )
#0 639.8   transformer_engine/jax/csrc/extensions/activation.cpp:41:46: error: non-constant-expression cannot be narrowed from type 'long' to 'unsigned long' in initializer list [-Wc++11-narrowing]
#0 639.8      41 |   auto output_shape = std::vector<size_t>{m, n};
#0 639.8         |                                              ^
#0 639.8   transformer_engine/jax/csrc/extensions/activation.cpp:41:46: note: insert an explicit cast to silence this issue
#0 639.8      41 |   auto output_shape = std::vector<size_t>{m, n};
#0 639.8         |                                              ^
#0 639.8         |                                              static_cast<unsigned long>( )
#0 639.8   transformer_engine/jax/csrc/extensions/activation.cpp:42:49: error: non-constant-expression cannot be narrowed from type 'long' to 'unsigned long' in initializer list [-Wc++11-narrowing]
#0 639.8      42 |   auto output_trans_shape = std::vector<size_t>{n, m};
#0 639.8         |                                                 ^
#0 639.8   transformer_engine/jax/csrc/extensions/activation.cpp:42:49: note: insert an explicit cast to silence this issue
#0 639.8      42 |   auto output_trans_shape = std::vector<size_t>{n, m};
#0 639.8         |                                                 ^
#0 639.8         |                                                 static_cast<unsigned long>( )
#0 639.8   transformer_engine/jax/csrc/extensions/activation.cpp:256:45: error: non-constant-expression cannot be narrowed from type 'long' to 'unsigned long' in initializer list [-Wc++11-narrowing]
#0 639.8     256 |   auto input_shape = std::vector<size_t>{m, n};
#0 639.8         |                                             ^
#0 639.8   transformer_engine/jax/csrc/extensions/activation.cpp:256:45: note: insert an explicit cast to silence this issue
#0 639.8     256 |   auto input_shape = std::vector<size_t>{m, n};
#0 639.8         |                                             ^
#0 639.8         |                                             static_cast<unsigned long>( )
#0 639.8   transformer_engine/jax/csrc/extensions/activation.cpp:257:49: error: non-constant-expression cannot be narrowed from type 'long' to 'unsigned long' in initializer list [-Wc++11-narrowing]
#0 639.8     257 |   auto act_input_shape = std::vector<size_t>{m, n * act_len};
#0 639.8         |                                                 ^~~~~~~~~~~
#0 639.8   transformer_engine/jax/csrc/extensions/activation.cpp:257:49: note: insert an explicit cast to silence this issue
#0 639.8     257 |   auto act_input_shape = std::vector<size_t>{m, n * act_len};
#0 639.8         |                                                 ^~~~~~~~~~~
#0 639.8         |                                                 static_cast<unsigned long>( )
#0 639.8   transformer_engine/jax/csrc/extensions/activation.cpp:258:46: error: non-constant-expression cannot be narrowed from type 'long' to 'unsigned long' in initializer list [-Wc++11-narrowing]
#0 639.8     258 |   auto output_shape = std::vector<size_t>{m, n * act_len};
#0 639.8         |                                              ^~~~~~~~~~~
#0 639.8   transformer_engine/jax/csrc/extensions/activation.cpp:258:46: note: insert an explicit cast to silence this issue
#0 639.8     258 |   auto output_shape = std::vector<size_t>{m, n * act_len};
#0 639.8         |                                              ^~~~~~~~~~~
#0 639.8         |                                              static_cast<unsigned long>( )
#0 639.8   transformer_engine/jax/csrc/extensions/activation.cpp:259:49: error: non-constant-expression cannot be narrowed from type 'long' to 'unsigned long' in initializer list [-Wc++11-narrowing]
#0 639.8     259 |   auto output_trans_shape = std::vector<size_t>{n * act_len, m};
#0 639.8         |                                                 ^~~~~~~~~~~
#0 639.8   transformer_engine/jax/csrc/extensions/activation.cpp:259:49: note: insert an explicit cast to silence this issue
#0 639.8     259 |   auto output_trans_shape = std::vector<size_t>{n * act_len, m};
#0 639.8         |                                                 ^~~~~~~~~~~
#0 639.8         |                                                 static_cast<unsigned long>( )
#0 639.8   transformer_engine/jax/csrc/extensions/activation.cpp:260:42: error: non-constant-expression cannot be narrowed from type 'long' to 'unsigned long' in initializer list [-Wc++11-narrowing]
#0 639.8     260 |   auto dbias_shape = std::vector<size_t>{n * act_len};
#0 639.8         |                                          ^~~~~~~~~~~
#0 639.8   transformer_engine/jax/csrc/extensions/activation.cpp:260:42: note: insert an explicit cast to silence this issue
#0 639.8     260 |   auto dbias_shape = std::vector<size_t>{n * act_len};
```

reproducer
```dockerfile
# syntax=docker/dockerfile:1.14.0@sha256:0232be24407cc42c983b9b269b1534a3b98eea312aad9464dd0f1a9e547e15a7
ARG uv_version=0.6.17
FROM ghcr.io/astral-sh/uv:${uv_version} AS uv_base
FROM nvcr.io/nvidia/cuda:12.9.1-cudnn-devel-ubuntu24.04
# FROM ghcr.io/nvidia/jax:jax-2025-07-06@sha256:d1faa4714a40ab512184fb471fc039972b71f82dd3370f377d31fbcc37fabfe4 AS base

# Enable bytecode compilation
# Copy from the cache instead of linking since it's a mounted volume
# Disable installer metadata, to create a deterministic layer.
ENV UV_COMPILE_BYTECODE=1 UV_NO_INSTALLER_METADATA=1 UV_LINK_MODE=copy
ENV UV_HOME="/opt/uv" UV_CACHE_DIR="/app/.cache" UV_PROJECT_ENVIRONMENT="/app/.venv"
ENV DEBIAN_FRONTEND=noninteractive
ENV PATH="/app/.venv/bin:/root/.local/bin:$PATH"

WORKDIR /app

# peripheral deps...not really essential to repro
RUN --mount=type=cache,target=/var/cache/apt <<EOF
apt-get update && \
apt-get install -y --no-install-recommends \
  build-essential=* \
  curl=* \
  git=* \
  libbz2-dev=* \
  libffi-dev=* \
  liblzma-dev=* \
  libncurses-dev=* \
  libreadline-dev=* \
  libsqlite3-dev=* \
  libssl-dev=* \
  llvm=* \
  openssh-client=* \
  openssh-server=* \
  python3-openssl=* \
  software-properties-common=* \
  tk-dev=* \
  wget=* \
  xz-utils=* \
  zlib1g-dev=* \
  netcat-traditional=*
EOF

RUN <<EOF
wget https://apt.llvm.org/llvm.sh
chmod +x llvm.sh
./llvm.sh 20
update-alternatives --install /usr/bin/clang clang /usr/bin/clang-20 60 --slave /usr/bin/clang++ clang++ /usr/bin/clang++-20
update-alternatives --install /usr/bin/cc cc /usr/bin/clang-20 60 --slave /usr/bin/cc++ cc++ /usr/bin/clang++-20
update-alternatives --install /usr/bin/c++ c++ /usr/bin/clang++-20 60
EOF

RUN mkdir -p /var/run/sshd

COPY --from=uv_base /uv /uvx /bin/

RUN mkdir -p ~/.ssh && chmod 0600 ~/.ssh && ssh-keyscan github.com >> ~/.ssh/known_hosts

COPY --from=uv_base /uv /uvx /bin/

RUN uv venv --seed --relocatable --python-preference only-managed --python 3.12

RUN --mount=type=ssh \
  --mount=type=cache,target=/root/.cache/pip \
  --mount=type=cache,target=/app/.cache \
  git clone --recursive https://github.com/NVIDIA/TransformerEngine.git \
  && cd TransformerEngine \
  && git submodule update --init --recursive \
  && MAX_JOBS=$(( $(nproc) / 2)) NVTE_FRAMEWORK=jax pip wheel . -v -w /wheels

```

seems like `size_t` vs `unsigned long` in the auto inferred types? casting to size_t seems potentially right for most cases? (cpp noob admittedly)

--- a/transformer_engine/jax/csrc/extensions/activation.cpp
+++ b/transformer_engine/jax/csrc/extensions/activation.cpp
@@ -37,9 +37,9 @@ Error_Type ActLuFFI(cudaStream_t stream, Buffer_Type input_buf, Buffer_Type scal
   auto is_2x = static_cast<bool>(is_2x_int);
   auto flatten_axis = output_buf->dimensions().size() - 1;  // output does not have act axis
 
-  auto input_shape = std::vector<size_t>{m, act_len * n};
-  auto output_shape = std::vector<size_t>{m, n};
-  auto output_trans_shape = std::vector<size_t>{n, m};
+  auto input_shape = std::vector<size_t>{m, static_cast<size_t>(act_len * n)};
+  auto output_shape = std::vector<size_t>{m, static_cast<size_t>(n)};
+  auto output_trans_shape = std::vector<size_t>{static_cast<size_t>(n), m};
   auto input_tensor = TensorWrapper(input, input_shape, static_cast<DType>(in_dtype));
   auto output_tensor = TensorWrapper(get_nvte_scaling_mode(scaling_mode));
   output_tensor.set_rowwise_data(output, static_cast<DType>(out_dtype), output_shape);
@@ -253,11 +253,11 @@ Error_Type DActLuDBiasQuantizeFFI(cudaStream_t stream, Buffer_Type input_buf,
   auto m = product(act_input_dims, 0, act_input_dims.size() - 2);
   auto n = input_dims.back();
 
-  auto input_shape = std::vector<size_t>{m, n};
-  auto act_input_shape = std::vector<size_t>{m, n * act_len};
-  auto output_shape = std::vector<size_t>{m, n * act_len};
-  auto output_trans_shape = std::vector<size_t>{n * act_len, m};
-  auto dbias_shape = std::vector<size_t>{n * act_len};
+  auto input_shape = std::vector<size_t>{m, static_cast<size_t>(n)};
+  auto act_input_shape = std::vector<size_t>{m, static_cast<size_t>(n * act_len)};
+  auto output_shape = std::vector<size_t>{m, static_cast<size_t>(n * act_len)};
+  auto output_trans_shape = std::vector<size_t>{static_cast<size_t>(n * act_len), m};
+  auto dbias_shape = std::vector<size_t>{static_cast<size_t>(n * act_len)};
   std::vector<size_t> workspace_shape(workspace_dims.begin(), workspace_dims.end());
 
   auto input_tensor =

--- a/transformer_engine/jax/csrc/extensions/normalization.cpp
+++ b/transformer_engine/jax/csrc/extensions/normalization.cpp
@@ -118,7 +118,7 @@ Error_Type NormForwardFFI(cudaStream_t stream, Buffer_Type x_buf, Buffer_Type sc
         convert_ffi_datatype_to_te_dtype(scale_inv_buf->element_type()),
         std::vector<size_t>{
             product(scale_inv_buf->dimensions(), 0, scale_inv_buf->dimensions().size() - 1),
-            scale_inv_buf->dimensions().back()});
+            static_cast<size_t>(scale_inv_buf->dimensions().back())});
   }
 
   if (scaling_mode == JAXX_Scaling_Mode::DELAYED_TENSOR_SCALING && is_fp8_dtype(out_dtype)) {
@@ -135,7 +135,7 @@ Error_Type NormForwardFFI(cudaStream_t stream, Buffer_Type x_buf, Buffer_Type sc
         convert_ffi_datatype_to_te_dtype(colwise_scale_inv_buf->element_type()),
         std::vector<size_t>{product(colwise_scale_inv_buf->dimensions(), 0,
                                     colwise_scale_inv_buf->dimensions().size() - 1),
-                            colwise_scale_inv_buf->dimensions().back()});
+                            static_cast<size_t>(colwise_scale_inv_buf->dimensions().back())});
   }
 
   if (_norm_type == NVTE_Norm_Type::LayerNorm) {

